### PR TITLE
docs: include notes about globals in migration-guide

### PR DESCRIPTION
**Pull request by Grohden**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

I'm trying to not get too much frustrated with the migration & migration guides.. so I thought I could contribute on the issues I have faced, hopefully they're welcome 🙏 

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Added a note about the required instalation of the `globals` package (docs don't [make it clear that globals is not a magic package](https://github.com/sindresorhus/globals/issues/244) and that we need to make it a direct dev dependency)

### CODE REVIEW

**Comment by aladdin-add** [comment]
> This was fixed 2 years ago, https://github.com/sindresorhus/globals/pull/184. Please upgrade your package.

https://github.com/sindresorhus/globals/issues/239#issuecomment-1968731224

When installing it, you will always get the latest version. This change may be more distracting to users.

**Comment by aladdin-add** [comment]
I would suggest a more appropriate place to be: https://eslint.org/docs/latest/use/troubleshooting/. Let's hear from the others. @eslint/eslint-team

**Comment by Grohden** [comment]
@aladdin-add 

> When installing it, you will always get the latest version. This change may be more distracting to users.

You have to remember that some devs (like me) are stupid 😅, I thought this was some magic global from node or something ("it worked before, I should just change the code to what the guide tells me right?"), so I just imported the package and saw that it was already installed in `node_modules`, so I thought that `eslint` included it as a dependency on its own to facilitate things, then I got that error.

About the troubleshooting:  I'm already trying to find a way to solve the global problem.. why do I have to hunt in another **potentially** unrelated place the solution? For the specific string search on docs (whitespace issue) I guess it makes sense to be there, but saying "hey, this is not magic, we included the package for you before, but now you'll need to install and use it"

**Comment by aladdin-add** [comment]
> About the troubleshooting: I'm already trying to find a way to solve the global problem.. why do I have to hunt in another potentially unrelated place the solution? For the specific string search on docs (whitespace issue) I guess it makes sense to be there, but saying "hey, this is not magic, we included the package for you before, but now you'll need to install and use it yourself" would be helpful

you should always install the package - even if you have not encountered this problem. It may be working without explicitly installing as it may be an indirect dependency, while  your package manager (like npm v3+) just flattened it. But it should be avoided.

File: docs/src/use/configure/migration-guide.md [hunk 1]
```diff
@@ -330,6 +330,8 @@ export default [
 ];
 ```
 
+Note that for this to work or if you get `Global "AudioWorkletGlobalScope " has leading or trailing whitespace`, you'll need to install the latest `globals` package.
```
Comment (nzakas) [inline docs/src/use/configure/migration-guide.md:333]: This is way too specific an error message to include here. We can also format this a bit nicer.

```suggestion
::: tip
You'll need to install the `globals` package from npm for this example to work. It is not automatically installed by ESLint.
:::
```

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**